src/EditorFeatures/CSharpTest/InlineMethod/CSharpInlineMethodTests.cs

```diff
@@ -0,0 +1,1751 @@
+public class TestClass
+{
+    public void Calller()
+    {
+        if ((1 | 2) is int i)
+        {
+        }
+    }
+
+    private int Callee()
+    {
+        return 1 | 2;
+    }
+}");
+
+        [Fact]
+        public Task TestAsExpression()
+            => TestVerifier.TestInRegularAndScript1Async(
+                @"
+public class TestClass
+{
+    public void Calller(bool f)
+    {
+        var x = Cal[||]lee(f) as string;
+    }
+
+    private object Callee(bool f)
+    {
+        return f ? ""Hello"" : ""World"";
+    }
+}",
+                @"
+public class TestClass
+{
+    public void Calller(bool f)
+    {
+        var x = (f ? ""Hello"" : ""World"") as string;
+    }
+
+    private object Callee(bool f)
+    {
+        return f ? ""Hello"" : ""World"";
+    }
+}");
+
+        [Fact]
+        public Task TestCoalesceExpression()
+            => TestVerifier.TestInRegularAndScript1Async(
+                @"
+public class TestClass
+{
+    public int Caller(int? c)
+    {
+        return 1 + Cal[||]lee(c);
+    }
+
+    private int Callee(int? c)
+    {
+        return c ?? 1;
+    }
+}",
+                @"
+public class TestClass
+{
+    public int Caller(int? c)
+    {
+        return 1 + (c ?? 1);
+    }
+
+    private int Callee(int? c)
+    {
+        return c ?? 1;
+    }
+}");
+
+        [Fact]
+        public Task TestCoalesceExpressionAsRightValue()
+            => TestVerifier.TestInRegularAndScript1Async(
+                @"
+public class TestClass
+{
+    public string Caller(string c)
+    {
+        return c ?? Cal[||]lee(null);
+    }
+
+    private string Callee(string c2)
+    {
+        return c2 ?? ""Hello"";
+    }
+}",
+                @"
+public class TestClass
+{
+    public string Caller(string c)
+    {
+        return c ?? null ?? ""Hello"";
+    }
+
+    private string Callee(string c2)
+    {
+        return c2 ?? ""Hello"";
+    }
+}");
+
+        [Fact]
+        public Task TestSimpleMemberAccessExpression()
+            => TestVerifier.TestInRegularAndScript1Async(
+                @"
+public class TestClass
+{
+    public void Caller()
+    {
+        var x = C[||]allee().Length;
+    }
+
+    private string Callee() => ""H"" + ""L"";
+}",
+                @"
+public class TestClass
+{
+    public void Caller()
+    {
+        var x = (""H"" + ""L"").Length;
+    }
+
+    private string Callee() => ""H"" + ""L"";
+}");
+
+        [Fact]
+        public Task TestElementAccessExpression()
+            => TestVerifier.TestInRegularAndScript1Async(
+                @"
+public class TestClass
+{
+    public void Caller()
+    {
+        var x = C[||]allee()[0];
+    }
+
+    private string Callee() => ""H"" + ""L"";
+}",
+                @"
+public class TestClass
+{
+    public void Caller()
+    {
+        var x = (""H"" + ""L"")[0];
+    }
+
+    private string Callee() => ""H"" + ""L"";
+}");
+
+        [Fact(Skip = "No Precedence support")]
+        public Task TestSwitchExpression()
```

Comment:
@CyrusNajmabadi  I am calling [this](https://github.com/dotnet/roslyn/blob/0d3cf92c300459e40dae31b09aa6c7f5001aaaa4/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ExpressionSyntaxExtensions.cs#L662) to get the precedence of the expression [here](https://github.com/Cosifne/roslyn/blob/c38554ce4d61fedab50ece9d63e029efcc851ead/src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs#L235) but found it seems doesn't have the switch expression and SuppressNullableWarningExpression. So I mark both test as skipped. (Looks like the API is only supporting all the expression of C# 7.3?)
Is there an other API to get all the precedence of the expression?